src/storage/cloud_storage/mod.rs

```diff
@@ -124,7 +129,11 @@ impl<U: Sync + Send> StorageBackend<U> for CloudStorage {
             .map_err(|_| Error::from(ErrorKind::PermanentFileNotAvailable))?;
         let response: Response<Body> = client.request(request).map_err(|_| Error::from(ErrorKind::PermanentFileNotAvailable)).await?;
         let body = unpack_response(response).await?;
-        Ok(Object::new(body.bytes().into()))
+        let start_pos: usize = usize::try_from(start_pos).map_err(|_| Error::from(ErrorKind::PermanentFileNotAvailable))?;
+        if body.bytes().len() < start_pos {
+            return Err(Error::from(ErrorKind::PermanentFileNotAvailable));
+        }
+        Ok(Object::new(body.bytes()[start_pos..].into()))
```

Comment:
We're gonna have to look at some point into how to do this without pulling everything into memory.